sycl/source/handler.cpp

```diff
@@ -206,10 +207,22 @@ event handler::finalize() {
         MHostKernel->call(
             MNDRDesc, (NewEvent) ? NewEvent->getHostProfilingInfo() : nullptr);
         return CL_SUCCESS;
+      } else {
```

Comment:
Please, don't use `else` after return.
https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#f56-avoid-unnecessary-condition-nesting

Reply:
Updated with single-point-of-return.